### ISSUE TITLE
unnecessary plotfinder.py stuff

### ISSUE THREAD

**Issue by almsam** [almsam/Numerical-Analysis-code#26]
the min, max, & derivative don't need to be calculayed in plotfinder.py - these methods should be casted to a new file

here is the unneccesary code

```

def input_function():
    func_str = input("Enter a function (e.g., '2*x**2 + 4*x + 1'): ")
    x = symbols('x')
    return sympify(func_str)

def evaluate_function(func, x_val):
    x = symbols('x')
    return float(func.subs(x, x_val))

def find_min_max(func, range_start, range_end):
    x = symbols('x')
    critical_points = solve(diff(func, x), x)
    valid_points = [p for p in critical_points if range_start <= p <= range_end]
    valid_points.extend([range_start, range_end])
    values = [evaluate_function(func, p) for p in valid_points]
    return min(values), max(values)

def calculate_derivative(func, x_val):
    x = symbols('x')
    deriv = diff(func, x)
    return float(deriv.subs(x, x_val))
```

```
    # User-defined function regression
    user_func = input_function()
    user_func_error = user_defined_regression(x, y, user_func)
    error_list.append(("User-defined", user_func_error))

    # Find min and max of user-defined function
    min_val, max_val = find_min_max(user_func, min(x), max(x))
    print(f"\nMin value of user-defined function: {min_val}")
    print(f"Max value of user-defined function: {max_val}")

    # Calculate derivative at mean x
    mean_x = np.mean(x)
    deriv_at_mean = calculate_derivative(user_func, mean_x)
    print(f"Derivative of user-defined function at mean x ({mean_x}): {deriv_at_mean}")
```

the following 2 blocks of code should remain in plotfinder, the rest of the above needs a new file

```

def input_function():
    func_str = input("Enter a function (e.g., '2*x**2 + 4*x + 1'): ")
    x = symbols('x')
    return sympify(func_str)

    # User-defined function regression
    user_func = input_function()
    user_func_error = user_defined_regression(x, y, user_func)
    error_list.append(("User-defined", user_func_error))


```

**Comment by almsam** [comment]
pushed to #27

**Comment by almsam** [comment]
merged in #27 & #28 - closed
